proto/dwtools/abase/l0.test/Long.test.s

```diff
@@ -13470,35 +13511,55 @@ function longFromRangeWithStep( test )
   test.case = 'wrong arguments';
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep( 'wrong arguments' ) );
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep( undefined ) );
-  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( null ) );
 
   test.case = 'without arguments';
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep() );
 
-  test.case = 'without step argument'
+  test.case = 'not enough arguments'
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 0, 1 ] ) );
 
-  test.case = 'without range argument'
-  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( 2 ) );
-
-  test.case = 'argument is not array';
-  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( 1, 3 ) );
-
   test.case = 'more then 2 arguments'
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 0, 1 ], 1, 2 ) );
 
-  test.case = 'argument is not a number'
-  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 0, 1 ], notnumber ) );
+  /**/
+
+  test.case = 'wrong type of range'
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( 'wrong', 2 ) );
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( 1, 3 ) );
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( undefined, 3 ) );
 
-  test.case = 'one of range is infinity'
+  test.case = 'one of range numbers is infinity'
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 1, Infinity ], 1 ) );
   test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ -Infinity, 1 ], 1 ) );
 
-  test.case = 'decrementing range with step > 0';
-  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 10, 0 ], 2 ) );
+  /**/
+
+  test.case = 'wrong type of step'
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 0, 1 ], 'wrong' ) );
+
+  test.case = 'range[ 0 ] > range[ 1 ], step is not a number'
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 0, 1 ], NaN ) );
+
+  test.case = 'range[ 0 ] > range[ 1 ], step is null'
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 0, 1 ], 0 ) );
+
+  test.case = 'range[ 0 ] > range[ 1 ], step is positive';
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 1, 0 ], 2 ) );
+
+  test.case = 'range[ 0 ] > range[ 1 ], step is positive float';
+  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 1, 0 ], 1.5 ) );
```

Comment:
Please, remove duplicates like this. 
```js
  test.case = 'range[ 0 ] > range[ 1 ], step is positive';
  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 1, 0 ], 2 ) );

  test.case = 'range[ 0 ] > range[ 1 ], step is positive float';
  test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 1, 0 ], 1.5 ) );
```
Float and Integer step does not matter for test case.